### PR TITLE
Reorganization

### PR DESCRIPTION
**Pull request by daler**
WIP. Still need to make sure the wrapper tests are working, and the multiqc wrapper needs some work in order for multiqc to put its output in a subdir (e.g., aggregation_dir/multiqc.html).

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
 gffutils ==0.8.7.1
 hisat2 ==2.0.5
 lcdblib ==0.1
-pysam ==0.9.1.4
+pandas ==0.19.2
+picard ==2.7.1
+pysam ==0.10.0
 pytest ==3.0.5
 samtools ==1.3.1
 snakemake ==3.10.1
```
Comment (jfear) [inline requirements.txt]: Go ahead and bump to 3.11.2

File: rnaseq.snakefile [hunk 2]
```diff
@@ -2,7 +2,7 @@ import os
 import pandas as pd
 from lcdblib.snakemake import helpers, aligners
 from lcdblib.utils import utils
-import common
+from lib import common
 
 shell.prefix('set -euo pipefail; ')
```
Comment (jfear) [inline rnaseq.snakefile]: I have been adding TMPDIR stuff here, I don't know if we want to added some code and just have it commented out:

```python
#TMPDIR = os.path.join('/lscratch', os.getenv('SLURM_JOBID'))
#shell.prefix("set -euo pipefail; export TMPDIR={};".format(TMPDIR))
```

File: rnaseq.snakefile [hunk 3]
```diff
@@ -19,33 +19,36 @@ samples = sampletable.ix[:, 0]
...
+        'bam': '{sample_dir}/{sample}/{sample}.cutadapt.markdups.bam',
+        'metrics': '{sample_dir}/{sample}/{sample}.cutadapt.markdups.bam.metrics',
     },
     'kallisto': {
-        'h5': 'samples/{sample}/{sample}/kallisto/abundance.h5',
+        'h5': '{sample_dir}/{sample}/{sample}/kallisto/abundance.h5',
     },
 }
-fill = dict(sample=samples, count=['.count', ''])
+fill = dict(sample=samples, count=['.count', ''], sample_dir=sample_dir, agg_dir=agg_dir)
```
Comment (jfear) [inline rnaseq.snakefile]: I don't see any use for `count`, I think it is old.

File: wrappers/requirements.txt [hunk 4]
```diff
@@ -0,0 +1,11 @@
+# only put tools that are needed for tests (e.g., `bowtie-build` is needed to make sure
+# that bowtie-index worked), not for the wrappers themselves, which should have
+# their own environment.yml files.
+bowtie2 ==2.3.0
+gffutils ==0.8.7.1
+hisat2 ==2.0.5
+lcdblib ==0.1
+pysam ==0.9.1.4
```
Comment (jfear) [inline wrappers/requirements.txt]: bump 0.10.0

File: wrappers/requirements.txt [hunk 5]
```diff
@@ -0,0 +1,11 @@
+# only put tools that are needed for tests (e.g., `bowtie-build` is needed to make sure
+# that bowtie-index worked), not for the wrappers themselves, which should have
+# their own environment.yml files.
+bowtie2 ==2.3.0
+gffutils ==0.8.7.1
+hisat2 ==2.0.5
+lcdblib ==0.1
```
Comment (jfear) [inline wrappers/requirements.txt]: bump 0.1.1

**Comment by jfear** [comment]
This is probably a bit OCD, but it makes me feel better! 

Do you have a standard of line breaks in a rule. For example:

```python
rule my_rule_w_break:
    input:
        r1='myin_r1.fq',
        r2='myin_r2.fq'
    output:
        bam='myout.bam'
    log:
        'myout.bam.log'
    params:
        extra='-s SMILE'
    wrapper:
        wrapper_for('smile')
```

vs:

```python
rule my_rule_no_break:
    input: 
        r1='myin_r1.fq',
        r2='myin_r2.fq'
    output: bam='myout.bam'
    log: 'myout.bam.log'
    params: extra='-s SMILE'
    wrapper: wrapper_for('smile')
```

For sure when a directive has more then one item it is cleaner to break them up on multiple lines. But for single directive arguments, especially things like `log` and `wrapper` it adds a lot of extra lines.

I go back and forth between the two. Usually when I have 10 or more rules I start to use no breaks just to make the file a little more scrollable. Just curious, maybe I am the only one OCD enough to think about this :-)

File: rnaseq.snakefile (hunk 2 above)
Comment (daler) [inline rnaseq.snakefile]: Good call, for now I'll do: 

```python
TMPDIR = tempfile.gettempdir()                                                                                                                                                                                                                                           
JOBID = os.getenv('SLURM_JOBID')                                                                                                                                                                                                                                         
if JOBID:                                                                                                                                                                                                                                                                
    TMPDIR = os.path.join('/lscratch', JOBID)                                                                                                                                                                                                                            
shell.prefix('set -euo pipefail; export TMPDIR={};'.format(TMPDIR))
```
 might be useful to add a `get_temp_dir()` to lcdblib or common.py

File: rnaseq.snakefile (hunk 3 above)
Comment (daler) [inline rnaseq.snakefile]: Good catch, thanks.

**Comment by daler** [comment]
Re: OCD your call . . . I'm fine with leaving it ad hoc or picking a policy (either one) and sticking with it, whatever you decide.

**Comment by daler** [comment]
:tada:

**Pull request merged**